include/Macro.h

```diff
@@ -248,21 +248,23 @@
 // bitwise indices of derived fields
 // --> start from (1L<<NCOMP_TOTAL) to distinguish from the intrinsic fields
 // --> remember to define NDERIVE = total number of derived fields
-#  define _VELX               ( 1L << (NCOMP_TOTAL+0) )
-#  define _VELY               ( 1L << (NCOMP_TOTAL+1) )
-#  define _VELZ               ( 1L << (NCOMP_TOTAL+2) )
-#  define _PRES               ( 1L << (NCOMP_TOTAL+3) )
-#  define _TEMP               ( 1L << (NCOMP_TOTAL+4) )
+#  define _VELX               ( 1L << (NCOMP_TOTAL+ 0) )
+#  define _VELY               ( 1L << (NCOMP_TOTAL+ 1) )
+#  define _VELZ               ( 1L << (NCOMP_TOTAL+ 2) )
+#  define _VELR               ( 1L << (NCOMP_TOTAL+ 3) )
+#  define _PRES               ( 1L << (NCOMP_TOTAL+ 4) )
+#  define _TEMP               ( 1L << (NCOMP_TOTAL+ 5) )
+#  define _EINT               ( 1L << (NCOMP_TOTAL+ 6) )
 # ifdef MHD
-#  define _MAGX_CC            ( 1L << (NCOMP_TOTAL+6) )
-#  define _MAGY_CC            ( 1L << (NCOMP_TOTAL+7) )
-#  define _MAGZ_CC            ( 1L << (NCOMP_TOTAL+8) )
-#  define _MAG_ENGY_CC        ( 1L << (NCOMP_TOTAL+9) )
-#  define _DERIVED            ( _VELX | _VELY | _VELZ | _PRES | _TEMP | _MAGX_CC | _MAGY_CC | _MAGZ_CC | _MAG_ENGY_CC )
-#  define NDERIVE             9
+#  define _MAGX_CC            ( 1L << (NCOMP_TOTAL+ 7) )
+#  define _MAGY_CC            ( 1L << (NCOMP_TOTAL+ 8) )
+#  define _MAGZ_CC            ( 1L << (NCOMP_TOTAL+ 9) )
+#  define _MAG_ENGY_CC        ( 1L << (NCOMP_TOTAL+10) )
+#  define _DERIVED            ( _VELX | _VELY | _VELZ | _VELR | _PRES | _TEMP | _EINT | _MAGX_CC | _MAGY_CC | _MAGZ_CC | _MAG_ENGY_CC )
+#  define NDERIVE             11
 # else
-#  define _DERIVED            ( _VELX | _VELY | _VELZ | _PRES | _TEMP )
-#  define NDERIVE             5
+#  define _DERIVED            ( _VELX | _VELY | _VELZ | _VELR | _PRES | _TEMP | _EINT )
+#  define NDERIVE             7
 # endif
 
```

Comment:
We will need to compute the radial profiles of "entropy", "Ye", "Ynu1/2" (neutrino fractions) and "Znu1/2" (neutrino entropies) when doing neutrino transport. 
Perhaps, adding an extra `#ifdef ` (if doing neutrino transport ...) statement.

Reply:
These will be supported automatically as long as the corresponding _XXX symbolic constants are defined in `include/Macro.h`. For the nuclear EoS in your `dev-ccsn` branch, there are already `_ENTR` and `_YE`. So the only thing we need to do is to add the neutrino entropies :)

Reply:
I should clarify that in the comment above I've assumed that `Entr`, `Ye`, and `Znu1/2` should be treated as passive scalars (since we have to evolve them by solving additional continuity equations) instead of derived fields. Right?

Reply:
Yes, `Ye`, `Ynu1`, `Ynu2`, `Znu1`, and `Znu2` are passive scalars. `Entr` could be a derived field.

Reply:
OK. We can add new derived fields in the ccsn branch